src/serializer/ResidualHeapSerializer.js

```diff
@@ -151,11 +151,16 @@ export class ResidualHeapSerializer {
     properties: Map<string, PropertyBinding> = obj.properties,
     objectPrototypeAlreadyEstablished: boolean = false
   ) {
-    /*
-    for (let symbol of obj.symbols.keys()) {
-      // TODO #22: serialize symbols
+    //inject symbols
+    for (let [symbol, propertyBinding] of obj.symbols) {
+      invariant(propertyBinding.descriptor);
+      let desc = propertyBinding.descriptor;
+      // TODO:#22 move to emit property
```

Comment:
It seems that Prepack already supports conditional updates of symbol properties.

Please include the following test case:
```js
function() {
    let x = global.__abstract ? __abstract("boolean", "(true)") : true;
    let obj = {};
    let symbol = Symbol();
    if (x) {
        obj[symbol] = 42;
    }
    inspect = function() { return obj[symbol]; }
})();
```